src/domain/interfaces/types.ts

```diff
@@ -36,8 +36,89 @@ export interface IPaymentTokenMapping {
     payeeIdType: PayeeIdType
 }
 
-export interface IHttpResponseData {
-    payload : unknown
+export type IHttpResponse = {
+    payload : Payee | Quote | Transfer
+}
+
+export type Transfer = {
+    completedTimestamp: string,
+    fulfilment: string,
+    homeTransactionId: string,
+    transferState: string
+}
+
+export type ITransferRequest = {
+    homeR2PTransactionId: string
+    amount: string,
+    amountType: string,
+    currency: string,
+    from: unknown,
+    ilpPacket: unknown,
+    note: string,
+    quote: Quote,
+    quoteRequestExtensions: string,
+    subScenario: string,
+    to: Payee,
+    transactionType: string,
+    transferId: string,
+    transactionRequestId: string
+}
+
+export type Quote = {
+    expiration: string,
+    extensionList: unknown[],
+    geoCode: unknown,
+    payeeFspCommissionAmount: string,
+    payeeFspCommissionAmountCurrency: string,
+    payeeFspFeeAmount: string,
+    payeeFspFeeAmountCurrency: string,
+    payeeReceiveAmount: string,
+    payeeReceiveAmountCurrency: string,
+    quoteId: string,
+    transactionId: string,
+    transferAmount: string,
+    transferAmountCurrency: string
+}
+
+export type Payee = {
+    dateOfBirth: string,
+    displayName: string,
+    extensionList: unknown[],
+    firstName: string,
+    fspId: string,
+    idSubValue: string,
+    idType: string,
+    idValue: string,
+    lastName: string,
+    merchantClassificationCode: string,
+    middleName: string,
+    type: string,
+    supportedCurrencies: string,
+    kycInformation: string
+}
+
+export type IQuoteRequestData = {
```

Comment:
These types are exported in api-snippets library. If you want you can directly use those types from there.

Reply:
Let me test this out.